### PR TITLE
Copy in right menu, notification and refactor

### PR DESCRIPTION
**Pull request by ketewan**
## Change Summary

Small fixes for cell copy logic:
1) Notification appears on successful copy
2) Copy option is added to the right-click context menu
3) Multiselection doesn't go away on right-click within the selected area
4) Clipboard content is a little bit more intellectual -- the table stays table for some other applications (because of html format)

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by ketewan** [comment]
@wingkwong I don't think the failed test is connected to my changes. Could you please rerun the workflow?

**Comment by wingkwong** [comment]
Got some conflicting files. Please rebase.

**Comment by ketewan** [comment]
> Got some conflicting files. Please rebase.

Done :)

File: packages/nc-gui/composables/useMultiSelect/cellRange.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+export type Cell =
+  | {
```
Comment (mustafapc19) [inline packages/nc-gui/composables/useMultiSelect/cellRange.ts:2]: Can be changed to 
```
export interface Cell {
  row: number | null
  col: number | null
}
```

File: packages/nc-gui/composables/useMultiSelect/copyValue.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { UITypes } from '../../../nocodb-sdk'
+import type { ColumnType } from '../../../nocodb-sdk'
+import type { Row } from '~/lib'
+
+export const copyTable = async (rows: Row[], cols: ColumnType[]) => {
```
Comment (mustafapc19) [inline packages/nc-gui/composables/useMultiSelect/copyValue.ts:5]: Nice! Content looks really neat on my notes app.

**Comment by mustafapc19** [comment]
Issue: 
- Page hangs if u have an active selection and select another table.

File: packages/nc-gui/composables/useMultiSelect/index.ts [hunk 3]
```diff
@@ -1,18 +1,16 @@
 import type { MaybeRef } from '@vueuse/core'
-import { UITypes } from 'nocodb-sdk'
-import { message, reactive, ref, unref, useCopy, useEventListener, useI18n } from '#imports'
-
-interface SelectedBlock {
-  row: number | null
-  col: number | null
-}
+import type { ColumnType } from '../../../nocodb-sdk'
```
Comment (wingkwong) [inline packages/nc-gui/composables/useMultiSelect/index.ts:2]:
```
import type { ColumnType } from 'nocodb-sdk'
```

File: packages/nc-gui/composables/useMultiSelect/index.ts [hunk 4]
```diff
@@ -26,99 +24,88 @@ export function useMultiSelect(
...
+          const rowObj = unref(data)[cpRow]
+          const columnObj = unref(fields)[cpCol]
+
+          const textToCopy = (columnObj.title && rowObj.row[columnObj.title]) || ''
+
+          await copy(textToCopy)
+          message.success(t('msg.info.copiedToClipboard'))
+        }
+      }
+    } catch {}
```
Comment (wingkwong) [inline packages/nc-gui/composables/useMultiSelect/index.ts:57]: any reason not to show error?

**Review by wingkwong — COMMENTED** [review]
You may check the copied values on LTAR / Geometry cells. Sometimes it works as expected but sometimes it shows `[object Object]`.

File: packages/nc-gui/composables/useMultiSelect/cellRange.ts (hunk 1 above)
Comment (ketewan) [reply packages/nc-gui/composables/useMultiSelect/cellRange.ts:2]: I actually meant here, that either both row and col can be null or none, that's not the same as what you wrote. But probably it's not very important and your variant is more readable.

**Review by wingkwong — COMMENTED** [review]
`[object Object]` issue again. Copying from a single cell is fine but it fails when copying from multiple cells including that.

e.g.

```
14033335568 	 [object Object] 	 2022-10-25 12:24:28
``` 

expected:

```
14033335568 	{"x":153.1408538,"y":-27.6333361} 	 2022-10-25 12:24:28
```

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 5]
```diff
@@ -208,48 +201,44 @@ const { selectCell, selectBlock, selectedRange, clearRangeRows, startSelectRange
...
     const rows = tbodyEl.value?.querySelectorAll('tr')
     const cols = rows?.[row].querySelectorAll('td')
     const td = cols?.[col === 0 ? 0 : col + 1]
-
     if (!td || !gridWrapper.value) return
-
     const { height: headerHeight } = tableHead.value!.getBoundingClientRect()
     const tdScroll = getContainerScrollForElement(td, gridWrapper.value, { top: headerHeight, bottom: 9, right: 9 })
-
     if (rows && row === rows.length - 2) {
```
Comment (pranavxc) [inline packages/nc-gui/components/smartsheet/Grid.vue:242]: Please bring back the spacing(newline) between these statements, that way it looks cleaner.

**Review by pranavxc — APPROVED** [review]

**Comment by o1lab** [comment]
Spacing suggestions will have it done.

**Pull request merged**